app/containers/MessageBox/RecordAudio.js

```diff
@@ -0,0 +1,201 @@
+		audioQuality: Audio.RECORDING_OPTION_IOS_AUDIO_QUALITY_MIN,
+		sampleRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.sampleRate,
+		numberOfChannels: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.numberOfChannels,
+		bitRate: Audio.RECORDING_OPTIONS_PRESET_LOW_QUALITY.ios.bitRate,
+		outputFormat: Audio.RECORDING_OPTION_IOS_OUTPUT_FORMAT_MPEG4AAC
+	}
+};
+const RECORDING_MODE = {
+	allowsRecordingIOS: true,
+	playsInSilentModeIOS: true,
+	staysActiveInBackground: false,
+	shouldDuckAndroid: true,
+	playThroughEarpieceAndroid: false,
+	interruptionModeIOS: Audio.INTERRUPTION_MODE_IOS_DO_NOT_MIX,
+	interruptionModeAndroid: Audio.INTERRUPTION_MODE_ANDROID_DO_NOT_MIX
+};
+
+const _formatTime = function(seconds) {
+	let minutes = Math.floor(seconds / 60);
+	seconds %= 60;
+	if (minutes < 10) { minutes = `0${ minutes }`; }
+	if (seconds < 10) { seconds = `0${ seconds }`; }
+	return `${ minutes }:${ seconds }`;
+};
+
+const startRecordingAudio = async(instance, setRecordingStatus, setRecorderBusy) => {
+	setRecorderBusy(true);
+	try {
+		const permissions = await Audio.getPermissionsAsync();
+
+		if (permissions.status === 'granted') {
+			await Audio.setAudioModeAsync(RECORDING_MODE);
+			instance.setOnRecordingStatusUpdate((status) => {
+				setRecordingStatus(status);
+			});
+
+			await instance.prepareToRecordAsync(RECORDING_SETTINGS);
+			await instance.startAsync();
+			activateKeepAwake();
+		} else {
+			await Audio.requestPermissionsAsync();
+		}
+	} catch (error) {
+		// Do nothing
+	}
+	setRecorderBusy(false);
+};
+
+const finishRecordingAudio = async(instance, onFinish, setRecorderBusy) => {
+	setRecorderBusy(true);
+	try {
+		await instance.stopAndUnloadAsync();
+
+		const fileURI = instance.getURI();
+		const fileData = await getInfoAsync(fileURI);
+		const fileInfo = {
+			name: `${ Date.now() }.aac`,
+			mime: 'audio/aac',
+			type: 'audio/aac',
+			store: 'Uploads',
+			path: fileURI,
+			size: fileData.size
+		};
+
+		onFinish(fileInfo);
+	} catch (error) {
+		// Do nothing
+	}
+	deactivateKeepAwake();
+	setRecorderBusy(false);
+};
+
+const cancelAudioMessage = async(instance, setRecorderBusy) => {
+	setRecorderBusy(true);
+	try {
+		await instance.stopAndUnloadAsync();
+	} catch (error) {
+		// Do nothing
+	}
+	deactivateKeepAwake();
+	setRecorderBusy(false);
+};
+
+const RecordAudio = ({ theme, recordingCallback, onFinish }) => {
+	const recordingInstance = useRef(null);
+
+	const [recordingStatus, setRecordingStatus] = useState({
+		canRecord: false,
+		isRecording: false,
+		durationMillis: 0,
+		isDoneRecording: false
+	});
+	const [recorderBusy, setRecorderBusy] = useState(false);
+
+	useEffect(() => () => {
+		if (recordingInstance.current) {
+			cancelAudioMessage(recordingInstance.current, setRecorderBusy);
+		}
+	}, []);
+
+	useEffect(() => {
+		recordingCallback(recordingStatus.isRecording);
+	}, [recordingStatus.isRecording]);
+
+	const recordingContent = recordingStatus.isRecording ? (
```

Comment:
This is a bad pattern.
It's a huge `if` condition to attribute a constant and then returning it.
Why not
```
if (condition) {
  return ...
}

return ...
```
?